### PR TITLE
feat(python): New Python runtime that will work with remote

### PR DESCRIPTION
**Pull request by tebeka**
This is a rewrite of the Python runtime to work with gRPC, this allows remote runtimes to work with the same Python runtime.

Putting it here for review, I'll shape up the code later.

### CODE REVIEW

File: proto/Makefile [hunk 1]
```diff
@@ -39,3 +39,30 @@ py-dist: py-deps
...
+		remote/remote.proto
+
+.PHONY: remote-go
+remote-go:
+	mkdir -p ../runtimes/pythonrt/pb
+	protoc \
+		-I remote \
+		--go_out=../runtimes/pythonrt/pb \
+		--go_opt=paths=source_relative \
+		--go-grpc_out=../runtimes/pythonrt/pb \
```
Comment (efiShtain) [inline proto/Makefile]: I think you should generate it next to the other protos in the proto directory

the remotert would ues the same protos as well

**Comment by efiShtain** [comment]
you have some references to your local env specific "autokitteh-miki.ngrok.dev"

can you change it to a more generic thing?

**Comment by tebeka** [comment]
> you have some references to your local env specific "autokitteh-miki.ngrok.dev"
> 
> can you change it to a more generic thing?

This is just for my local testing. Changed.

File: runtimes/pythonrt/runner/Dockerfile
Comment (efiShtain) [inline runtimes/pythonrt/runner/Dockerfile:1]: do we need a docker file here ?

File: runtimes/pythonrt/runner/remote_pb2.py
Comment (efiShtain) [inline runtimes/pythonrt/runner/remote_pb2.py:1]: what's remote_pb2 vs remote_pb ?

File: runtimes/pythonrt/runner/Dockerfile
Comment (tebeka) [reply runtimes/pythonrt/runner/Dockerfile:1]: Was for testing, deleted.

File: proto/Makefile (hunk 1 above)
Comment (tebeka) [reply proto/Makefile]: Done.

File: runtimes/pythonrt/runner/remote_pb2.py
Comment (tebeka) [reply runtimes/pythonrt/runner/remote_pb2.py:1]: There's only `remote_pb2*` files, that's what Python's gRPC generates.

File: runtimes/pythonrt/svc.go [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+}
+
+func freePort() (int, error) {
+	conn, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return 0, err
+	}
+
+	conn.Close()
+	return conn.Addr().(*net.TCPAddr).Port, nil
```
Comment (efiShtain) [inline runtimes/pythonrt/svc.go:288]: if you close the connection, how can you be sure the port is still free ?

File: runtimes/pythonrt/svc.go [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+}
+
+// Start starts the server on a free port in a new goroutine.
+// It returns the port the server listens on.
+func (s *remoteSvc) Start() error {
+	port, err := freePort()
```
Comment (efiShtain) [inline runtimes/pythonrt/svc.go:294]: not sure you need this freePort call

just use 
```go
lis, err := net.Listen("tcp", addr)
if err != nil {
 	return err
 }
s.lis = lis
s.port = lis.Addr().(*net.TCPAddr).Port
```

File: runtimes/pythonrt/svc.go [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+
+	if syscallFn.IsValid() {
+		svc.syscallFn = syscallFn
+	}
+
+	return &svc
+}
+
+func (s *remoteSvc) Health(context.Context, *pb.HealthRequest) (*pb.HealthResponse, error) {
+	return &pb.HealthResponse{}, nil
```
Comment (efiShtain) [inline runtimes/pythonrt/svc.go:65]: I'm thinking about checking with temporal here if the workflow is still active

wdyt ?
let's say someone canceled the workflow, or there was a worker restart somehow, not sure

File: runtimes/pythonrt/svc.go (hunk 3 above)
Comment (efiShtain) [reply runtimes/pythonrt/svc.go:294]: do we really need to start a new server for each python process ?

we can add this capability to ak itself, and use the runner ID to get back to the correct go routine of the workflow itself

should be less resources, less error prone (I think)

**Comment by tebeka** [comment]
Closing this since merged from @efiShtain 's branch.

**Pull request closed without merging**